### ISSUE TITLE
orfeo thermal noise removal introduces nodata

### ISSUE THREAD

**Issue by jdries** [Open-EO/openeo-geopyspark-driver#293]

**Comment by jdries** [comment]
we can try setting -outputs.default to some value other than 0, hoping that denoising than still uses 0

**Comment by JeroenVerstraelen** [comment]
@jdries Which value other than 0 would be best for this?

**Comment by jdries** [comment]
not sure what datatype is used? Maybe set a high value, like 10000?

**Comment by JeroenVerstraelen** [comment]
It looks like the result dtype is always float32.

**Comment by jdries** [comment]
output is backscatter, which is usually a small decimal number, so any high floating point number should work.
Do you get output already?

**Comment by JeroenVerstraelen** [comment]
Yes, I am getting output. I'm just working on a better example test now, I've been using test_s1backscatter_orfeo.py::test_creodias_s1_backscatter for now. The noise removal is a bit strange because it both sets a good amount of noise pixels to NoData (=0), but also changes the [min, max] of the output from [0.000706709, 4.4512] to [1.20978e-08, 4.44767] which I assume is what you mean with "thermal noise is set to a very low value instead".

I found the detailed version of the outputs.default parameter description in the cli:
```
        -outputs.default         <float>          Default pixel value  (optional, off by default, default value is 0)
                                                  Default value to write when outside of input image
```
Am I correct that the issue is that noise_removal sets pixels close to 0, so we want pixels outside of the input image to be different from 0 so we can distinguish them?

**Comment by jdries** [comment]
yes, and I'm also hoping that noise removal does not use the nodata value set by outputs.default at all, so that is to be tested.

**Comment by JeroenVerstraelen** [comment]
Reverted the usage of outputs.default until the orfeo issue is fixed.

**Comment by JeroenVerstraelen** [comment]
Response on https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/issues/2329 
`Waiting for the fix I recommend you to call ManageNoData after the orthorectification, which is made for your use case.`

**Comment by soxofaan** [comment]
hmm https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/issues/2329   has been closed now:
> Closed because author of issue didn't provide needed information

**Comment by JeroenVerstraelen** [comment]
I did several experiments using the ManageNoData process to solve the bug that we noticed with `outputs.default`. First of all, the bug can be solved by first setting the NoDataValue of the SARCalibration to 0 and then to 10000. This will ensure that both the original NoData values and the pixels that lie outside of the original image have the same value.

However, isues arise when you want to ensure that the SARCalibration uses 10000 as NoData pixels since the orfeo implementation does not take these NoData pixels into account. You can get around this by saving a mask of the original NoData pixels, then performing SARCalibration, and then using the mask on the result. This way pixels that have a very low value or are zero are seen as noise, and pixels that have a value of 10000 are NoData. 

TODO: Finish comment




https://gist.github.com/JeroenVerstraelen/38a36fe097e5b9af96dc8e2996936796
